util/diff_runs.py

```diff
@@ -0,0 +1,244 @@
+            specBefore = '%s@%s' % (self.args.before.platforms[i],
+                                    self.args.before.sha)
+            specAfter = '%s@%s' % (self.args.after.platforms[i],
+                                   self.args.after.sha)
+            runBefore = self.fetcher.fetchResults(
+                beforeSHA, self.args.before.platforms[i])
+            runAfter = self.fetcher.fetchResults(
+                afterSHA, self.args.after.platforms[i])
+
+            if runBefore is None:
+                self.logger.warning('Failed to fetch %s' % specBefore)
+            if runAfter is None:
+                self.logger.warning('Failed to fetch %s' % specAfter)
+            if runBefore is None or runAfter is None:
+                continue
+
+            differences = 0
+            checked = 0
+            added = 0
+            removed = 0
+
+            for k in filter(lambda x: x not in runAfter, runBefore.keys()):
+                removed += 1
+
+            for test in runAfter.keys():
+                checked += 1
+
+                if test not in runBefore:
+                    added += 1
+                    continue
+
+                resultAfter = runAfter[test]
+                resultBefore = runBefore[test]
+
+                passingDelta = resultAfter[0] - resultBefore[0]
+                totalDelta = resultAfter[1] - resultBefore[1]
+
+                if totalDelta > 0:
+                    self.logger.info('%s has %d new tests (total)'
+                                     % (test, totalDelta))
+                elif totalDelta < 0:
+                    self.logger.info('%s has %d removed tests (total)')
+
+                if passingDelta < 0:
+                    self.logger.warning('%s has %d new failures'
+                                        % (test, math.fabs(passingDelta)))
+                elif passingDelta > 0:
+                    self.logger.info('%s has %d new passes'
+                                     % (test, passingDelta))
+
+                delta = math.fabs(totalDelta) + math.fabs(passingDelta)
+                logging.debug('%s has %d differences (%s vs %s)'
+                              % (test,
+                                 passingDelta,
+                                 resultAfter,
+                                 resultBefore))
+                differences += delta
+
+            self.logger.info(
+                'Finished diff of %s and %s with %d differences in %d tests'
+                % (specBefore, specAfter, differences, checked))
+            if (added > 0):
+                self.logger.info('%d tests ran in %s but not in %s'
+                                 % (added, specAfter, specBefore))
+            if (removed > 0):
+                self.logger.info('%d tests ran in %s but not in %s'
+                                 % (removed, specBefore, specAfter))
+
+
+class Fetcher(object):
+    '''Placeholder for stubbing 'fetchResults' for the unit tests.'''
```

Comment:
I don't think I understand this comment.

Reply:
Updated.